### PR TITLE
Task#any_assignable_to? should return a boolean

### PR DESCRIPTION
**Pull request by shanear**
Fixes prop warnings:

screen shot 2017-04-13 at 5 25 14 pm

### CODE REVIEW

**Comment by kierachell** [comment]
**MASTER**

screen shot 2017-04-13 at 8 31 24 pm


**THIS BRANCH**
screen shot 2017-04-13 at 8 33 13 pm

File: app/models/task.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def assign_next_to!(user)
     end
 
     def any_assignable_to?(user)
-      user.current_task(self) || next_assignable
+      !(user.current_task(self).nil? && next_assignable.nil?)
```
Comment (joofsh) [inline app/models/task.rb]: This is really hard for me to read. Was the original logic correct just it needs to be cast to a boolean? 

Why not just `!!` at the beginning? Negating nil is awkward imo

Comment (shanear) [inline app/models/task.rb]: i agree. I was trying to appease the ever-vengeful Rubocop 🤖. I will create a tech-improvement story to get rid of that rule

Comment (kierachell) [inline app/models/task.rb]: I ❤️ janky logic;

isn't that why we go to school!?

**Pull request merged**